src/com/google/javascript/jscomp/RewriteDynamicImports.java

```diff
@@ -357,4 +375,82 @@ private void injectWrappingFunctionExtern() {
     externsRoot.addChildToBack(wrappingFunctionDefinition);
     compiler.reportChangeToEnclosingScope(wrappingFunctionDefinition);
   }
+
+  /**
+   * A shallow traversal class for the externs to inject the alias.
+   *
+   * Later passes require the names to be defined. For simple name aliases first check
+   * for an existing definition. Qualified name aliases check for an existing definition of the
+   * root name only.
+   */
+  private class AliasInjectingTraversal extends NodeTraversal.AbstractPreOrderCallback {
+    @Override
+    public boolean shouldTraverse(NodeTraversal t, Node n, Node parent) {
+      checkState(aliasIsValid());
+
+      // If the extern is a simple name and already declared, there is nothing to do
+      if (NodeUtil.isValidSimpleName(alias) && t.getScope().getVar(alias) != null) {
+        return false;
+      }
+
+      // Create the type of the import function
+      JSTypeRegistry registry = compiler.getTypeRegistry();
+      JSType unknownType = registry.getNativeType(JSTypeNative.UNKNOWN_TYPE);
+      JSType promiseType = registry.getNativeType(JSTypeNative.PROMISE_TYPE);
+      TemplatizedType promiseTemplatizedType =
+          registry.createTemplatizedType(promiseType.toObjectType(), ImmutableList.of(unknownType));
+      JSType stringType = registry.getNativeType(JSTypeNative.STRING_TYPE);
+
+      Node externsRoot = compiler.getSynthesizedExternsInputAtEnd().getAstRoot(compiler);
```

Comment:
Is this a typo?  I see the compiler has a `getSynthesizedExternsInput` but no `getSynthesizedExternsInputAtEnd`

Reply:
It had a `getSynthesizedExternsInputAtEnd` method when the PR was opened - but there's been a refactoring since then. I've updated the methods.